tsconfig.json

```diff
@@ -1,22 +1,27 @@
 {
   "compilerOptions": {
-    "baseUrl": "src",
-    "rootDir": "./",
     "outDir": "build",
-    "target": "esnext",
+    "target": "es5",
+    "lib": [
+      "dom",
+      "dom.iterable",
+      "esnext"
```

Comment:
Why do you have `esnext` in the `lib` section?

Reply:
I guess we might not need it in this lib we could go down a version but I did "esnext" for since we were using BigInt in Xpring-JS. I'm not sure what the harm is in using esnext it doesn't impact build size just what we are allowed to use in the source code.

Reply:
Sure, this is probably fine.

I'm unhappy not because of the code you wrote, but because this is the least well documented part of the TypeScript compiler settings. It's undocumented what `libs` are included by default when your target is `es2017`, the use case for `lib` is pretty much undocumented... I just don't like configuration settings I don't understand well.

But we are using `BigInt`. That does mean that if `xpring-common-js` _is_ used in the browser, they may need to polyfill `BigInt`: https://caniuse.com/#search=bigint

Reply:
If we omitted the lib entry the default lib for target of ES2017 would be `DOM,ES2017,DOM.Iterable`

When working with Lib's I refer to the source code to see what each version gives us. 
https://github.com/Microsoft/TypeScript/tree/master/lib
*esnext is es2020*

Yes, we do need to polyfill bigInt this is one of the PR's I will have out today actually. Once this is polyfilled we should probably just use the default lib options for the ES2017 so that we will see errors when using things that should probably be polyfilled.

BigInt is only used in Xpring-JS so I can actually make that change now on this Library.